RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -201,26 +203,28 @@ void MahiFit::doFit(std::array<float, 3>& correctedOutput, int nbx) const {
   }
 }
 
-double MahiFit::minimize() const {
+const float MahiFit::minimize() const {
   nnlsWork_.invcovp.setZero(nnlsWork_.tsSize, nnlsWork_.nPulseTot);
   nnlsWork_.ampVec.setZero(nnlsWork_.nPulseTot);
-  nnlsWork_.aTaMat.setZero(nnlsWork_.nPulseTot, nnlsWork_.nPulseTot);
-  nnlsWork_.aTbVec.setZero(nnlsWork_.nPulseTot);
+
+  SampleMatrix invCovMat;
+  invCovMat.setConstant(nnlsWork_.tsSize, nnlsWork_.tsSize, nnlsWork_.pedVal);
+  invCovMat += nnlsWork_.noiseTerms.asDiagonal();
 
   double oldChiSq = 9999;
```

Comment:
Still ```double```, but they are assigned to a ```float``` later on: they can be also defined as```float```

Reply:
fixed